src/trace/trace.c

```diff
@@ -497,6 +503,28 @@ void trace_off(void)
 	k_spin_unlock(&trace->lock, key);
 }
 
+void trace_print_version_banner(void)
+{
+	/*
+	 * Non-Zephyr builds emit the version banner in DMA-trace
+	 * init and this is done at a later time as otherwise the
+	 * banner would be lost. With Zephyr logging subsystem in use,
+	 * we can simply print the banner at boot.
+	 *
+	 * META_QUOTE(SOF_SRC_HASH) is part of the format string so it
+	 * is part of log dictionary meta data and does not go to
+	 * the firmware binary (in case dictionary logging is used).
+	 * The value printed to log will be different from
+	 * SOF_SRC_HASH in case of mismatch.
+	 */
+#ifdef __ZEPHYR__
+	LOG_INF("FW ABI 0x%x DBG ABI 0x%x tags SOF:" SOF_GIT_TAG " zephyr:" \
+		META_QUOTE(BUILD_VERSION) " src hash 0x%08x (ref hash " \
+		META_QUOTE(SOF_SRC_HASH) ")",
+		SOF_ABI_VERSION, SOF_ABI_DBG_VERSION, SOF_SRC_HASH);
+#endif
+}
+
 void trace_init(struct sof *sof)
```

Comment:
CONFIG_TRACE=n
CONFIG_LOG=y
CONFIG_ZEPHYR_LOG=y

is this scenario valid? Because it wont print in such case
Not sure if we need to compile trace files with zephyr logs?

Reply:
@abonislawski Currently if you set CONFIG_TRACE=n , you'll only get Zephyr side logs, SOF side logs will be omitted via include/sof/trace/trace.h . 

It kind of makes sense not to print the SOF version.

Combination:
CONFIG_TRACE=n
CONFIG_ZEPHYR_LOG=y

is not correct though as ZEPHYR_LOG will have (currently) impact only if CONFIG_TRACE is selected.

Reply:
yeap but actually this is good because there is no point in compiling all trace stuff if we want to call zephyr LOG_ anyway.
So can you move this somewhere else print this without trace too?

Reply:
> I was assuming CONFIG_TRACE ifdef covers all the trace macros, but it doesn't.

The best recipe to debug pre-processing is the same as the one for debugging build system issues: **break it** on purpose.

So you're not sure whether some code / macro is used in some specific CONFIG_uration? Easy: add a syntax error in the source. If it still compiles then you know for sure it's not used. This is literally a million times faster than reading the source or trying to understand the output of `gcc -E`.

Even better: when it fails to compile the compiler shows you the actual stack of macros, something literally impossible to figure out by just reading the source past some macro nesting and Kconfig complexity.

It's a bit like test-driven development where the only way to "test the test" is to (re-)introduce a bug.

Error handling in compilers is seriously underrated. There's a basic but very informative and super easy to use source code analyzer hidden in compiler error messages.